core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1037,6 +1038,8 @@ class ReplicaManager(val config: KafkaConfig,
         errorReadingData = true
       if (logReadResult.divergingEpoch.nonEmpty)
         hasDivergingEpoch = true
+      if (logReadResult.preferredReadReplica.getOrElse(config.brokerId) != config.brokerId)
```

Comment:
I think that we could use `logReadResult.preferredReadReplica.nonEmpty` because it is only set when a follower as been selected.

Reply:
done